woocommerce/woocommerce-ios
# Partial swipe back gesture from product details removes the content

Closes: #13651


## Description


There was an issue with swipe back gesture from product details. If you do not finish the gesture, basically do a "peak" to the screen before, the content would be removed.

After some debugging I noticed that we were using `navigationController(_ navigationController: UINavigationController, willShow viewController: UIViewController, animated: Bool)`, emphasis on **willShow** part. The content resetting part was done in there but the problem was that we should have used **didShow** version of it, to be sure that the back gesture was finished, that the back navigation was finished.

## Testing information


- Open Woo app on iPhone (device or simulator)
- Open Products tab
- Tap on one of the products
- Use the swipe back gesture partially, do not swipe back to products list completely
- Check that the product details are shown still
- Use the swipe back gesture again but do it completely
- Check that the products list is shown

## Screenshots


Before:

https://github.com/user-attachments/assets/51aefbdd-4339-4d63-92e5-41aa590a92ce

After:

https://github.com/user-attachments/assets/28be8ca3-f21d-4670-9391-21c723509b57

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
